plugins/modules/win_pssession_configuration.ps1

```diff
@@ -203,6 +203,155 @@ function ConvertFrom-AnsibleOption {
     }
 }
 
+function ConvertTo-PsNative {
+    <#
+    .SYNOPSIS
+    Normalize runtime objects to PowerShell-native types.
+
+    .DESCRIPTION
+    Semantics:
+    - Unwraps PSObject to BaseObject
+    - Converts IDictionary -> Hashtable (recursively) with string keys
+    - Converts IEnumerable -> array of parsed elements (recursively), preserving strings
+
+    .NOTES
+    - This is used for Ansible argument spec entries declared as "type='raw'"
+        so that module parameters arriving as CLR or PSObject-wrapped values
+        are made PowerShell-native before being passed to PowerShell cmdlets.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        $Value
+    )
+
+    End {
+        if ($null -eq $Value) { return $null }
+
+        # Unwrap PSObject wrapper if present
+        if ($Value -is [System.Management.Automation.PSObject] -and $Value.BaseObject -ne $null) {
+            $Value = $Value.BaseObject
+        }
+
+        # IDictionary -> Hashtable
+        if ($Value -is [System.Collections.IDictionary]) {
+            $ht = @{}
+            foreach ($k in $Value.Keys) {
+                $v = $Value[$k]
+                if ($k -is [string]) { $keyName = $k } else { $keyName = $k.ToString() }
+                $ht[$keyName] = ConvertTo-PsNative -Value $v
+            }
+            return $ht
+        }
+
+        # IEnumerable (but not string) -> array of parsed elements
+        if ($Value -is [System.Collections.IEnumerable] -and -not ($Value -is [string])) {
+            $out = @()
+            foreach ($e in $Value) { $out += (ConvertTo-PsNative -Value $e) }
+            return $out
+        }
+
+        return $Value
+    }
+}
+
+function ConvertTo-PsList {
+    <#
+    .SYNOPSIS
+    Normalize runtime objects into a PowerShell array of native elements.
+
+    .DESCRIPTION
+    Derived from the C# ParseList implementation in Ansible.Basic.cs.
+
+    .NOTES
+    - This is used for Ansible argument spec entries declared as "type='list'"
+        with "elements='raw'" so that module parameters arriving as CLR or
+        PSObject-wrapped values are made PowerShell-native before being passed
+        to PowerShell cmdlets.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        $Value
+    )
+
+    End {
+        if ($null -eq $Value) { return $null }
+        # Port of C# ParseList: perform .NET type checks to mirror behavior exactly.
+        $valueType = $Value.GetType()
+
+        # We'll collect parsed elements first, then coerce each element into the
+        # final shape (string or Hashtable) expected by New-PSSessionConfigurationFile.
+        $out = @()
```

Comment:
It's not great to do `+=` on an array. While the sizes we are dealing with won't really matter we can just do

```powershell
$out = @(
    if ($Value -is ...) {
         ... emit objects like normal
    }
    elseif ($Value ...) {
        ... More emitting
    }
)
```

PowerShell will collect all the output and coerce into an array all for you.